### PR TITLE
ci(client): Fix client trigger on any edit to root files

### PR DESCRIPTION
**Pull request by tylerbutler**
Unfortunately the change in #22133 is causing the client build to always be triggered for PRs, which was not the intended behavior.

Instead of `*`, let's try `*.*`, which I have seen used in other systems. The docs are not clear at all about the behavior.

I also reordered entries to be alphabetical and made folder entries consistent.

### CODE REVIEW

**Comment by jason-ha** [comment]
@tylerbutler, is this obsolete? It has a conflict; otherwise, I'd be okay approving.

**Comment by tylerbutler** [comment]
> @tylerbutler, is this obsolete? It has a conflict; otherwise, I'd be okay approving.

@jason-ha I reverted the original change, so the problem was mitigated, but this is still a correct change, I think. I missed the merge conflict. Resolved and should be ready to go now.

File: tools/pipelines/build-client.yml [hunk 1]
```diff
@@ -60,6 +60,8 @@ trigger:
     - release/*
   paths:
     include:
+    # Trigger on any edits to files in the root
+    - "*.*"
     - .prettierignore
     - biome.json
     - biome.jsonc
```
Comment (jason-ha) [inline tools/pipelines/build-client.yml:67]: Why do we need to list `biome.json` and other root files directly, too?

Comment (tylerbutler) [reply tools/pipelines/build-client.yml:67]: Hmmm, looks like a bad merge resolution on my part. One moment...

Comment (tylerbutler) [reply tools/pipelines/build-client.yml]: Root file entries are now removed, the list is alphabetized and normalized as was done in the original reverted change (#22133).

**Review by zhenmichael — COMMENTED** [review]
Would this pattern trigger on changes to "tools/pipelines" or the root? If it targets the tools directory, would a relative path like `../../*.*` work?

**Comment by tylerbutler** [comment]
> Would this pattern trigger on changes to "tools/pipelines" or the root?

It won't trigger on any edit in tools/pipelines - but the ones that are listed individually will trigger it.

>If it targets the tools directory, would a relative path like `../../*.*` work?

Not sure I follow. The intent is that this change causes the pipeline to trigger on changes to any file in the repo root - and no other behavior should change. Changes to individually listed files like the workflow files themselves will continue to trigger.

**Review by jason-ha — APPROVED** [review]

**Pull request merged**